core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -93,6 +114,44 @@ protected LogLevel failedResponseLogLevel() {
         return failedResponseLogLevel;
     }
 
+    /**
+     * Sets the {@link Function} to use when mapping the log level of request logs.
+     */
+    public T requestLogLevelMapper(Function<? super RequestLog, LogLevel> requestLogLevelMapper) {
+        if (this.isSetRequestLogLevel) {
+            throw new IllegalStateException("Request log level was already set.");
+        }
+        this.requestLogLevelMapper = requireNonNull(requestLogLevelMapper, "requestLogLevelMapper");
+        this.isSetRequestLogLevelMapper = true;
+        return self();
+    }
+
+    /**
+     * Returns the {@link LogLevel} to use when logging request logs.
+     */
+    protected Function<? super RequestLog, LogLevel> requestLogLevelMapper() {
+        return requestLogLevelMapper;
+    }
+
+    /**
+     * Sets the {@link Function} to use when mapping the log level of response logs.
+     */
+    public T responseLogLevelMapper(Function<? super RequestLog, LogLevel> responseLogLevelMapper) {
+        if (this.isSetSuccessfulResponseLogLevel || this.isSetFailedResponseLogLevel) {
+            throw new IllegalStateException("Response log level was already set.");
```

Comment:
Could you use different message for each case?

- `successfulResponseLogLevel has been set already.`
- `failedResponseLogLevel has been set already.`